flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaFieldTypeExtractor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.file.schema;
+
+import org.apache.flink.table.types.logical.RowType;
+
+import java.io.Serializable;
+import java.util.List;
+
+/** Extractor of schema for different tables. */
+public interface SchemaFieldTypeExtractor extends Serializable {
+    /**
+     * Extract key type from table schema.
+     *
+     * @param schema the table schema
+     * @return the key type
+     */
+    RowType keyType(TableSchema schema);
+
+    /**
+     * Extract key fields from table schema.
+     *
+     * @param schema the table schema
+     * @return the key fields
+     */
+    List<DataField> keyFields(TableSchema schema);
```

Comment:
We can just have util to convert `List<DataField>` to `RowType`?

Reply:
Do you mean the `RowType keyType (TableSchema schema)` and `List < DataField > keyFields (TableSchema schema)` methods in `Extractor`? These two methods are really a bit strange, we can change `RowType keyType (TableSchema schema)` to `RowType keyType (List < DataField > keyFields)`

Reply:
`(RowType) new RowDataType(false, fields).logicalType` this can convert `fields` to `RowType`.

Reply:
Yes, we can convert `List<DataField>` to `RowType` with `(RowType) new RowDataType(false, fields).logicalType` for `ChangelogValueCountFileStoreTable`, but for `ChangelogWithKeyFileStoreTable` we should do `addKeyNamePrefix` for the key `RowType`. 

So I added `keyType` in `Extractor`, `ChangelogValueCountFileStoreTable` and `ChangelogWithKeyFileStoreTable` can implement their own operations in it and `KeyValueFileStoreScan` can get key type for different tables

Reply:
So can we add this prefix to `List<DataField>` too? And document this in method comments.
I think this is confuse when I see two methods: `keytype` and `keyfields`.